app/modules/auth/__init__.py

```diff
@@ -26,7 +20,6 @@ def init_app(app, **kwargs):
     Init auth module.
     """
     # Bind Flask-Login for current_user
-    login_manager.request_loader(load_user_from_request)
```

Comment:
Given you drop this, you may as well drop the function. How does this work now, though?

Reply:
I actually did more work in my other project on this - I think I got login manager working as follows:
```
from flask import g
from flask.sessions import SecureCookieSessionInterface
from flask_login import user_loaded_from_header
from flask_login import LoginManager as OriginalLoginManager

class CustomSessionInterface(SecureCookieSessionInterface):
    """Prevent creating session from API requests."""
    def save_session(self, *args, **kwargs):
        if g.get('login_via_header'):
            return
        return super(CustomSessionInterface, self).save_session(*args,
                                                                **kwargs)


@user_loaded_from_header.connect
def user_loaded_from_header(self, user=None):
    g.login_via_header = True


class LoginManager(OriginalLoginManager):
    def init_app(self, app, add_context_processor=True):
        app.session_interface = CustomSessionInterface()
        super().init_app(app, add_context_processor=add_context_processor)

# in some other code 
from app.extensions import login_manager
from authlib.flask.oauth2 import current_token

@login_manager.request_loader
def load_user_from_request(request):
    """
    Load user from OAuth2 Authentication header.
    """
    if current_token:
        user = current_token.user
        if user:
            return user
        user_id = current_token.user.id
        if user_id:
            return User.query.get(user_id)
    return None

```